test/OpenTelemetry.Tests/Trace/ParentBasedSamplerTests.cs

```diff
@@ -105,47 +105,28 @@ public void DoNotExamineLinks()
     [InlineData(false, false)]
     public void CustomSamplers(bool parentIsRemote, bool parentIsSampled)
     {
-        var mockRepository = new MockRepository(MockBehavior.Strict);
-        var remoteParentSampled = mockRepository.Create<Sampler>();
-        var remoteParentNotSampled = mockRepository.Create<Sampler>();
-        var localParentSampled = mockRepository.Create<Sampler>();
-        var localParentNotSampled = mockRepository.Create<Sampler>();
+        var remoteParentSampled = new TestSampler();
+        var remoteParentNotSampled = new TestSampler();
+        var localParentSampled = new TestSampler();
+        var localParentNotSampled = new TestSampler();
 
         var samplerUnderTest = new ParentBasedSampler(
             new AlwaysOnSampler(), // root
-            remoteParentSampled.Object,
-            remoteParentNotSampled.Object,
-            localParentSampled.Object,
-            localParentNotSampled.Object);
+            remoteParentSampled,
+            remoteParentNotSampled,
+            localParentSampled,
+            localParentNotSampled);
 
         var samplingParams = MakeTestParameters(parentIsRemote, parentIsSampled);
-
-        Mock<Sampler> invokedSampler;
-        if (parentIsRemote && parentIsSampled)
-        {
-            invokedSampler = remoteParentSampled;
-        }
-        else if (parentIsRemote && !parentIsSampled)
-        {
-            invokedSampler = remoteParentNotSampled;
-        }
-        else if (!parentIsRemote && parentIsSampled)
-        {
-            invokedSampler = localParentSampled;
-        }
-        else
-        {
-            invokedSampler = localParentNotSampled;
-        }
-
         var expectedResult = new SamplingResult(SamplingDecision.RecordAndSample);
-        invokedSampler.Setup(sampler => sampler.ShouldSample(samplingParams)).Returns(expectedResult);
-
         var actualResult = samplerUnderTest.ShouldSample(samplingParams);
 
-        mockRepository.VerifyAll();
+        Assert.Equal(parentIsRemote && parentIsSampled, remoteParentSampled.LatestSamplingParameters.Equals(samplingParams));
```

Comment:
@ngruson I am finding these asserts bit confusing. Why cannot we simply assert like 

`Assert.True(remoteParentSampled.LatestSamplingParameters.Equals(samplingParams))`

Reply:
It depends on the inline data whether or not this assert is true.
If both params are true, the assert is true, otherwise false.

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/3188323/0233519f-b713-4e1b-a772-75645a397aeb)

Reply:
got it. An alternate way could be to use if statements, but that would make the test bit longer. Ok to keep it as is.